mist-common/src/main/java/edu/snu/mist/common/sources/EventGeneratorImpl.java

```diff
@@ -84,7 +126,27 @@ public OutputEmitter getOutputEmitter() {
   /**
    * If there is any remainder to do during start in downstream class, conduct it.
    */
-  protected abstract void startRemain();
+  protected void startRemain() {
+    if (checkpointPeriod != 0) {
+      checkpointResult = scheduler.scheduleAtFixedRate(new Runnable() {
+        public void run() {
+          final long checkpointTimestamp = getCurrentTimestamp() - expectedDelay;
+          outputEmitter.emitWatermark(
```

Comment:
We should not generate the watermark if the source uses PunctuatedWatermarkGenerator.